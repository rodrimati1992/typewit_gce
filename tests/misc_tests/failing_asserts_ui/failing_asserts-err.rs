#![feature(generic_const_exprs)]
#![allow(incomplete_features)]

use typewit_gce::const_marker::U8;

const fn aaa(_: u8) -> u8 { 3 }
const fn bbb(_: u8) -> u8 { 5 }


const fn foo<const X: u8, const Y: u8>() 
where
    U8<{Y * (3 + X)}>:,
    U8<{Y * (5 + X)}>:,
    U8<{X * X}>:,
    U8<{1 / X}>:,
    U8<{(1 / X) * (1 / X)}>:,
    U8<{1 % X}>:,
    U8<{(1 % X) * (1 % X)}>:,
{
    _ = typewit_gce::gce_int_eq!(3u8, 5);

    _ = typewit_gce::gce_int_eq!(Y * (3 + X), Y * (5 + X));

    // uses power syntax
    _ = typewit_gce::gce_int_eq!(aaa(3) * aaa(3), 0);
    _ = typewit_gce::gce_int_eq!(X * X, 0);
    _ = typewit_gce::gce_int_eq!(1/X, 0);
    _ = typewit_gce::gce_int_eq!((1/X) * (1/X), 0);
    _ = typewit_gce::gce_int_eq!(1 % X, 0);
    _ = typewit_gce::gce_int_eq!((1 % X) * (1 % X), 0);

    _ = typewit_gce::gce_int_eq!(aaa(3) / self::bbb(5), aaa(3) / (2 + self::bbb(5)));

    _ = typewit_gce::gce_int_eq!(aaa(3) % self::bbb(5), aaa(3) % (2 + self::bbb(5)));

    _ = typewit_gce::gce_int_eq!({ aaa(3) + 1 }, { bbb(3) + 1 });
}


fn main() {}