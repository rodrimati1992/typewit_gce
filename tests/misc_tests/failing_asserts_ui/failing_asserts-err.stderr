error: Cannot prove that the arguments are equal.
       This is their normalized representation:
       left: `3`
       right: `5`
  --> tests/misc_tests/failing_asserts_ui/failing_asserts-err.rs:20:9
   |
20 |     _ = typewit_gce::gce_int_eq!(3u8, 5);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Cannot prove that the arguments are equal.
       This is their normalized representation:
       left: `X * Y + 3 * Y`
       right: `X * Y + 5 * Y`
  --> tests/misc_tests/failing_asserts_ui/failing_asserts-err.rs:22:9
   |
22 |     _ = typewit_gce::gce_int_eq!(Y * (3 + X), Y * (5 + X));
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Cannot prove that the arguments are equal.
       This is their normalized representation:
       left: `aaa(3)**2`
       right: `0`
  --> tests/misc_tests/failing_asserts_ui/failing_asserts-err.rs:25:9
   |
25 |     _ = typewit_gce::gce_int_eq!(aaa(3) * aaa(3), 0);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Cannot prove that the arguments are equal.
       This is their normalized representation:
       left: `X**2`
       right: `0`
  --> tests/misc_tests/failing_asserts_ui/failing_asserts-err.rs:26:9
   |
26 |     _ = typewit_gce::gce_int_eq!(X * X, 0);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Cannot prove that the arguments are equal.
       This is their normalized representation:
       left: `1 / X`
       right: `0`
  --> tests/misc_tests/failing_asserts_ui/failing_asserts-err.rs:27:9
   |
27 |     _ = typewit_gce::gce_int_eq!(1/X, 0);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Cannot prove that the arguments are equal.
       This is their normalized representation:
       left: `(1 / X)**2`
       right: `0`
  --> tests/misc_tests/failing_asserts_ui/failing_asserts-err.rs:28:9
   |
28 |     _ = typewit_gce::gce_int_eq!((1/X) * (1/X), 0);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Cannot prove that the arguments are equal.
       This is their normalized representation:
       left: `1 % X`
       right: `0`
  --> tests/misc_tests/failing_asserts_ui/failing_asserts-err.rs:29:9
   |
29 |     _ = typewit_gce::gce_int_eq!(1 % X, 0);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Cannot prove that the arguments are equal.
       This is their normalized representation:
       left: `(1 % X)**2`
       right: `0`
  --> tests/misc_tests/failing_asserts_ui/failing_asserts-err.rs:30:9
   |
30 |     _ = typewit_gce::gce_int_eq!((1 % X) * (1 % X), 0);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Cannot prove that the arguments are equal.
       This is their normalized representation:
       left: `aaa(3) / self::bbb(5)`
       right: `aaa(3) / (2 + self::bbb(5))`
  --> tests/misc_tests/failing_asserts_ui/failing_asserts-err.rs:32:9
   |
32 |     _ = typewit_gce::gce_int_eq!(aaa(3) / self::bbb(5), aaa(3) / (2 + self::bbb(5)));
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Cannot prove that the arguments are equal.
       This is their normalized representation:
       left: `aaa(3) % self::bbb(5)`
       right: `aaa(3) % (2 + self::bbb(5))`
  --> tests/misc_tests/failing_asserts_ui/failing_asserts-err.rs:34:9
   |
34 |     _ = typewit_gce::gce_int_eq!(aaa(3) % self::bbb(5), aaa(3) % (2 + self::bbb(5)));
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Cannot prove that the arguments are equal.
       This is their normalized representation:
       left: `{ aaa (3 ) +1  }`
       right: `{ bbb (3 ) +1  }`
  --> tests/misc_tests/failing_asserts_ui/failing_asserts-err.rs:36:9
   |
36 |     _ = typewit_gce::gce_int_eq!({ aaa(3) + 1 }, { bbb(3) + 1 });
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `typewit_gce::gce_int_eq` (in Nightly builds, run with -Z macro-backtrace for more info)
